Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,282 @@
+  * [workload status subresource proposal](202409-status-subresource.md)
+
+> This proposal outlines the implementation of a Status subresource field extension to the Prometheus operator's configuration-based Custom Resource Definitions (CRDs).
+
+## Why
+
+The solution will allow users to verify whether their configurations have been successfully applied to the corresponding workload resources.
+
+Mapping between configuration resources and their associated workload resources:
+
+| Configuration Resource | Workload Resource              |
+|------------------------|--------------------------------|
+| ServiceMonitor         | Prometheus and PrometheusAgent |
+| PodMonitor             | Prometheus and PrometheusAgent |
+| Probes                 | Prometheus and PrometheusAgent |
+| ScrapeConfig           | Prometheus and PrometheusAgent |
+| PrometheusRule         | Prometheus and ThanosRuler     |
+| AlertmanagerConfig     | Alertmanager                   |
+
+## Pitfalls of the current solution
+
+Prometheus operator allows users to define their observability workloads through "workload" resources like `Prometheus`, `PrometheusAgent`, `AlertManager`. The configuration of these workloads can be done dynamically by orchestrating "configuration" resources like `ServiceMonitor`, `PodMonitor`, `ScrapeConfig`, etc.
+
+Currently, the status subresource is only implemented for workload resources. The absence of the status subresource for configuration resources makes it difficult to determine the source of the generated configuration of the workload resources. Additionally, there is no straightforward way to observe the reconciliation status of configuration resources. While Kubernetes events are available when a configuration is rejected by a workload, they are not sufficient for ongoing visibility or troubleshooting.
+
+## Goals
+
+* Define the structure of the status subresource for the custom resource definitions
+  * `ServiceMonitor`
+  * `PodMonitor`
+  * `ScrapeConfig`
+  * `Probes`
+  * `PrometheusRule`
+  * `AlertmanagerConfig`
+* Provide information about the targets being scraped and their status for scrape resources (`PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`).
+* Report when a configuration resource is considered invalid during reconciliation. For example:
+  * Feature not being supported by the version of the workload.
+  * Invalid configmap/secret key reference.
+  * Invalid PromQL expression in PrometheusRule resources.
+
+## Non-Goals
+
+* The solution does not aim to expose the full live configuration or runtime status of Prometheus. For example:
+  * It will not include per-target scrape status, only summary information (e.g., number of targets up/down).
+  * It will not surface fired alerts from PrometheusRule resources, as querying Prometheus for this data can be expensive and places undue load on both Prometheus and the operator.
```

Comment:
(nit)
```suggestion
  * It will not surface fired alerts from PrometheusRule resources, as querying Prometheus for this data can be expensive and places undue load on Prometheus, the operator and the Kubernetes platform in general.
```
I think that the last part could in the preamble

````
The solution does not aim to expose the full live configuration or runtime status of Prometheus. Doing so would be expensive for the workload, the operator and the Kubernetes system in general.
```